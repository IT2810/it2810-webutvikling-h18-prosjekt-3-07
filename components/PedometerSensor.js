import Expo from "expo";
import React from "react";
import { Pedometer } from "expo";
import { Text, View, StyleSheet } from "react-native";

export default class PedometerSensor extends React.Component {
  state = {
    isPedometerAvailable: "checking",
    pastStepCount: 0,
    currentStepCount: 0
  };

  componentDidMount() {
    this._subscribe();
  }

  componentWillUnmount() {
    this._unsubscribe();
  }

 // Subscribe to pedometer updates
  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentStepCount: result.steps
      });
    });

    //Check if pedometer is available.
    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerAvailable: String(result)
        });
      },
      error => {
        this.setState({
          isPedometerAvailable: "Could not get isPedometerAvailable: " + error
        });
      }
    );

    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - 1);
    // Set start and end date for step count. Is currently 24 hours. 
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastStepCount: result.steps });
      },
      error => {
        this.setState({
          pastStepCount: "Could not get stepCount: " + error
        });
      }
    );
  };

  _unsubscribe = () => {
    this._subscription && this._subscription.remove();
    this._subscription = null;
  };

  render() {
    const recomendedSteps = 10000;
    return (
      <View>
        <Text style={styles.text}>
          Steps taken in the last 24 hours: {this.state.pastStepCount + this.state.currentStepCount}
        </Text>
        <Text>You have: {recomendedSteps - (this.state.pastStepCount + this.state.currentStepCount)} steps left</Text>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  text: {
    color: 'red'

  }
});


Expo.registerRootComponent(PedometerSensor);