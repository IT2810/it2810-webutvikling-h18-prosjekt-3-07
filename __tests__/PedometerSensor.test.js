import React from "react";
import renderer from "react-test-renderer";
import PedometerSensor from "../components/PedometerSensor";

it("renders correctly", () => {
  const tree = renderer.create(<PedometerSensor />).toJSON();
  expect(tree).toMatchSnapshot();
});
