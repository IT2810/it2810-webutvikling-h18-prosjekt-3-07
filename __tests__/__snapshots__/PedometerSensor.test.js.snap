// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "marginTop": 20,
      }
    }
  >
    You have walked 
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
       
      0
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "normal",
      }
    }
  >
    steps today.
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "marginTop": 20,
        }
      }
    >
      Keep going! You have
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      10000
       
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "normal",
          }
        }
      >
         steps left 🏃‍♀️
      </Text>
    </Text>
  </View>
</View>
`;
